plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts

```diff
@@ -45,12 +45,17 @@ export class KubernetesClientProvider {
     };
 
     const kc = new KubeConfig();
-    kc.loadFromOptions({
-      clusters: [cluster],
-      users: [user],
-      contexts: [context],
-      currentContext: context.name,
-    });
+    if (clusterDetails.serviceAccountToken) {
```

Comment:
Hm. Does this need to be updated now, after https://github.com/backstage/backstage/pull/9046 so that it checks for `googleServiceAccount` as well? Or am I mixing stuff up

Reply:
hi! This one will use RBAC auth.

and  googleServiceAccount will use GCP roles (GCP IAM).

So still needed?

Reply:
I am not sure, I think I'll let you decide. Should `loadFromOptions` or `loadFromDefault` be called on GCP? I'm happy to merge after just settling that last question :)

Reply:
:) sure. 

`loadFromOptions` or `loadFromDefault` will be called on GKE. Line 48 was using the **Kubernetes service account** token.  The permission is defined by ClusterRole and RoleBindings.

#9046 add the auth provider use **Google Service account.**  token. The access is controlled by the GCP IAM roles using workload identities on cluster not RBAC. 

So basically, they both have same name called service account but KSA and GSA are looking for different credentials.